python/lsst/ts/pmd/component.py

```diff
@@ -127,13 +127,15 @@ def send_msg(self, msg):
         self.log.debug(f"Message to be sent is {msg}")
         self.commander.write(f"{msg}\r".encode())
         self.log.debug("Message written")
-        try:
-            reply = self.commander.read_until(b"\r")
+        reply = self.commander.read_until(b"\r")
+        # Hub returns an empty string if a device is not read successfully
+        # instead of raising a timeout exception
+        if reply != b"":
```

Comment:
Isn't this condition always true since by line 130 `reply` always at least contains `b"\r"`?

Reply:
Not in this case, as the hub returns an empty byte string b"" when a device does not return a value. So I check that the reply is not an empty string.

Reply:
But then line 130 will never complete. The commander is read until the string ends with `b"\r"` and immediately after you check for `b""`. Those two cannot co-exist.

Reply:
Right, and reading the commander will time out. Now I get it.